pkg/networkmanager/firewall.jsx

```diff
@@ -378,10 +378,10 @@ class AddEditServicesModal extends React.Component {
         }
         p.then(() => Dialogs.close())
                 .catch(error => {
-                    this.setState({
-                        dialogError: this.state.custom ? _("Failed to add port") : _("Failed to add service"),
+                    this.setState(prevState => ({
+                        dialogError: prevState.custom ? _("Failed to add port") : _("Failed to add service"),
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18387-20230222-094250-b66008dd-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)